src/store/flow/resource.ts

```diff
@@ -28,48 +27,55 @@ import {IFlowsState} from '@/store/flow/index'
 import {IRootState} from '@/store'
 
 export const getters: GetterTree<IFlowsState, IRootState> = {
-  resourcesByUuid: ({resources}) => keyBy(resources, 'uuid'),
+  resourcesByUuidOnActiveFlow: (_state, getters) => keyBy(getters.activeFlow.resources, 'uuid'),
 
-  resourceUuidsOnActiveFlow: (state, getters) => filter(map(getters.activeFlow.blocks, (block) => block.config.prompt)),
-
-  resourcesOnActiveFlow: (state, getters): IResource[] => filter(
-    state.resources,
-    (resource) => includes(getters.resourceUuidsOnActiveFlow, resource.uuid)
-  ),
+  resourceUuidsOnActiveFlow: (_state, getters) => filter(map(getters.activeFlow.resources, (res) => res.uuid)),
 }
 
-export const mutations: MutationTree<IFlowsState> = {
-  resource_add({resources}, {resource}: { resource: IResource }) {
-    resources.push(resource)
+export const mutations: MutationTree<IFlowsState> = {}
+
+export const actions: ActionTree<IFlowsState, IRootState> = {
+  resource_add({getters}, {resource}: {resource: IResource}) {
+    if (resource == null) {
+      throw new ValidationException('Unable to add null resource to flow')
+    }
+
+    getters.activeFlow.resources.push(resource)
   },
 
   // currently unused - see todo
-  resource_delete({resources}, {resourceId}: { resourceId: string }) {
+  resource_delete({getters}, {resourceId}: {resourceId: IResource['uuid']}) {
+    const flow: IFlow = getters.activeFlow
```

Comment:
Again, using a getter seems weird to me. It's not how things are ever done in the docs and I'm concerned it might have unintended consequences. From the docs:

> Actions are similar to mutations, the differences being that:
>
> Instead of mutating the state, actions commit mutations.
> Actions can contain arbitrary asynchronous operations.

So that suggests that actions should not change the state directly and then the fact that mutations don't have access to getters is suggestive of this being bad practice.

See this stackoverflow answer:

https://stackoverflow.com/questions/43258793/accessing-getters-within-vuex-mutations

Reply:
@jcbashdown , I agree on this! thanks for pointing this out. After spending hours on making flow-builder working again, I forgot this notion. 

Pls review again the code.

Reply:
Thanks for making these changes! All looks good now.